hbase-server/src/main/java/org/apache/hadoop/hbase/wal/RecoveredEditsOutputSink.java

```diff
@@ -89,20 +89,17 @@ private RecoveredEditsWriter getRecoveredEditsWriter(TableName tableName, byte[]
 
   @Override
   public List<Path> close() throws IOException {
-    boolean isSuccessful = true;
-    try {
-      isSuccessful = finishWriterThreads();
-    } finally {
-      isSuccessful &= closeWriters();
-    }
+    boolean isSuccessful;
+    isSuccessful = finishWriterThreads();
```

Comment:
I think finishWriterThreads will not close the writers, so we still need to close them even if finishWriterThreads returns false or throws exception.

So I think we should introduce something like abortWriters, if finishWriterThreads returns false or throws exception, we just close the writer, without renaming the wal file.

Thanks.